virtual_rainforest/core/registry.py

```diff
@@ -67,156 +73,75 @@ def register_module(module_name: str) -> None:
 
     # Extract the last component of the module name to act as unique short name
     module_name_short = module_name.split(".")[-1]
-    is_not_core = module_name != "virtual_rainforest.core"
+    if module_name_short in MODULE_REGISTRY:
+        LOGGER.warning(f"Module already registered: {module_name_short}")
+        return
 
-    LOGGER.info(f"Registering {module_name} module components")
+    is_core = module_name == "virtual_rainforest.core"
 
-    # Create an entry for the module
-    if module_name_short not in MODULE_REGISTRY:
-        MODULE_REGISTRY[module_name_short] = {}
+    LOGGER.info(f"Registering {module_name} module components")
 
-    # Register the model unless this is core
-    if is_not_core:
-        register_model(
-            module=module, module_name=module_name, module_name_short=module_name_short
+    # Check on the model argument.
+    if is_core and model is not None:
+        msg = "No model should be registered for the core module."
+        LOGGER.critical(msg)
+        raise RuntimeError(msg)
+    elif not is_core and model is None:
+        msg = "A model class is required to register model modules."
+        LOGGER.critical(msg)
+        raise RuntimeError(msg)
+    elif model is not None:
+        if module_name_short != model.model_name:
+            msg = f"The model_name attribute and module name differ in {module_name}"
+            LOGGER.critical(msg)
+            raise RuntimeError(msg)
```

Comment:
You don't need `else`/`elif` if the preceding block raises an exception and I find all the `elif`s make things less readable. I'd also personally prefer `if model` to `if model is not None`, but that is partly a question of taste.

```suggestion
    if is_core and model is not None:
        msg = "No model should be registered for the core module."
        LOGGER.critical(msg)
        raise RuntimeError(msg)
    if not is_core and model is None:
        msg = "A model class is required to register model modules."
        LOGGER.critical(msg)
        raise RuntimeError(msg)
    if model is not None and module_name_short != model.model_name:
        msg = f"The model_name attribute and module name differ in {module_name}"
        LOGGER.critical(msg)
        raise RuntimeError(msg)
```

Reply:
Can't merge those last two tests - the last clause doesn't just raise that exception but otherwise 👍